lib/iris/__init__.py

```diff
@@ -197,14 +197,18 @@ def __repr__(self):
         return msg.format(self.cell_datetime_objects, self.netcdf_promote,
                           self.netcdf_no_unlimited, self.clip_latitudes)
 
-    deprecated_options = {}
+    deprecated_options = {'cell_datetime_objects': 'warning'}
 
     def __setattr__(self, name, value):
         if name in self.deprecated_options:
-            reason = self.deprecated_options[name]
-            msg = ("the 'Future' object property {!r} is now deprecated. "
-                   "Please remove code which uses this.  {}")
-            warn_deprecated(msg.format(name, reason))
+            level = self.deprecated_options[name]
+            if level == 'error':
```

Comment:
Let's not over-engineer this yet. I'd simply not have the ability to set the "level" until we need it (and would therefore make deprecated_options a set literal)

Reply:
True. I have had my fingers in too many deprecation-removing pies and I was thinking ahead.